misk/src/main/kotlin/misk/client/HttpClientsConfig.kt

```diff
@@ -125,45 +85,30 @@ data class HttpClientConfig(
   val readTimeout: Duration? = null,
   val pingInterval: Duration? = null,
   val callTimeout: Duration? = null,
-  val maxRequests: Int = 128,
-  val maxRequestsPerHost: Int = 32,
-  val maxIdleConnections: Int = 100,
-  val keepAliveDuration: Duration = Duration.ofMinutes(5),
+  val maxRequests: Int? = null,
+  val maxRequestsPerHost: Int? = null,
+  val maxIdleConnections: Int? = null,
+  val keepAliveDuration: Duration? = null,
   val ssl: HttpClientSSLConfig? = null,
-  val unixSocketFile: File? = null,
+  val unixSocketFile: String? = null,
   val protocols: List<String>? = null
 )
 
-@Deprecated("Use default constructor")
-fun HttpClientEndpointConfig(
-  url: String? = null,
-  envoy: HttpClientEnvoyConfig? = null,
-  connectTimeout: Duration? = null,
-  writeTimeout: Duration? = null,
-  readTimeout: Duration? = null,
-  pingInterval: Duration? = null,
-  callTimeout: Duration? = null,
-  maxRequests: Int = 128,
-  maxRequestsPerHost: Int = 32,
-  maxIdleConnections: Int = 100,
-  keepAliveDuration: Duration = Duration.ofMinutes(5),
-  ssl: HttpClientSSLConfig? = null
-) = HttpClientEndpointConfig(
-    url = url,
-    envoy = envoy,
-    clientConfig = HttpClientConfig(
-        connectTimeout,
-        writeTimeout,
-        readTimeout,
-        pingInterval,
-        callTimeout,
-        maxRequests,
-        maxRequestsPerHost,
-        maxIdleConnections,
-        keepAliveDuration,
-        ssl
+fun HttpClientConfig.applyDefaults(other: HttpClientConfig) =
```

Comment:
Can we make this `internal`?

Reply:
Not sure. I probably prefer not making it internal given that HttpClientConfig is not internal and this function is basically composition of those configs. So in theory, if somebody wants to use HttpClientConfig directly - they will find this function useful.